node/node.go

```diff
@@ -392,13 +410,47 @@ func (n *Node) RefreshOnchainState(ctx context.Context) error {
 		select {
 		case <-ticker.C:
 			n.Logger.Info("Refreshing onchain state")
+			existingBlobParams := n.BlobVersionParams.Load()
 			blobParams, err := n.Transactor.GetAllVersionedBlobParams(ctx)
-			if err != nil {
+			if err == nil {
+				if existingBlobParams == nil || !existingBlobParams.Equal(blobParams) {
+					n.BlobVersionParams.Store(v2.NewBlobVersionParameterMap(blobParams))
+				}
+			} else {
 				n.Logger.Error("error fetching blob params", "err", err)
+			}
+
+			existingRelayClient, ok := n.RelayClient.Load().(clients.RelayClient)
+			if !ok {
+				n.Logger.Error("error fetching relay client")
+				continue
+			}
+
+			existingURLs := map[v2.RelayKey]string{}
+			if existingRelayClient != nil {
+				existingURLs = existingRelayClient.GetSockets()
+			}
+			relayURLs, err := n.Transactor.GetRelayURLs(ctx)
+			if err != nil {
+				n.Logger.Error("error fetching relay URLs", "err", err)
+				continue
+			}
+
+			if reflect.DeepEqual(existingURLs, relayURLs) {
+				n.Logger.Info("No change in relay URLs")
+				continue
+			}
+
+			relayClient, err := clients.NewRelayClient(&clients.RelayClientConfig{
```

Comment:
Relay url may be updated much more frequently than protocol params, should it be in a different interval/thread?

Reply:
Yeah, I agree in terms of relative frequency, but I still think the frequency for new relays being added and getting used would be in order of hours if not days.